tools/slang-embed/slang-embed.cpp

```diff
@@ -63,47 +63,119 @@ struct App
 {
     char const* appName = "slang-embed";
     char const* inputPath = nullptr;
-    char const* includeDir = nullptr;
     char const* outputPath = nullptr;
+    Slang::List<Slang::String> includeDirs;
     Slang::HashSet<Slang::String> includedFiles;
+    size_t charCount = 0;
+    bool useNewStringLit = true;
 
     void parseOptions(int argc, char** argv)
     {
-        // Options are currently all specified by position,
-        // so the parsing logic is simplistic.
-
+        // First, get the program name
         if (argc > 0)
         {
             appName = *argv++;
             argc--;
         }
 
-        if (argc > 0)
+        // Parse remaining arguments - we need at least inputPath
+        if (argc < 1)
         {
-            inputPath = *argv++;
-            argc--;
+            fprintf(stderr, "usage: %s inputPath [outputPath] [-I<includeDir> ...]\n", appName);
+            exit(1);
         }
 
-        if (argc > 0)
-        {
-            includeDir = *argv++;
-            argc--;
-        }
+        // Get input path (first positional argument)
+        inputPath = *argv++;
+        argc--;
 
-        if (argc > 0)
+        // Process remaining arguments
+        while (argc > 0)
         {
-            outputPath = *argv++;
+            char* arg = *argv++;
             argc--;
+
+            // Check for -I prefix for include directories
+            if (strncmp(arg, "-I", 2) == 0)
+            {
+                // Check if this is a concatenated string of include directories
+                char* startPtr = arg;
+
+                // Process the entire string as potentially multiple -I directives
+                while (startPtr && *startPtr)
+                {
+                    // Find the -I prefix
+                    char* iPos = strstr(startPtr, "-I");
+                    if (!iPos)
+                        break;
+
+                    // Move past the -I
+                    char* dirStart = iPos + 2;
+
+                    // Find the next -I or end of string
+                    char* nextIPos = strstr(dirStart, "-I");
+
+                    // Determine end of current include dir
+                    char* dirEnd = nextIPos ? nextIPos : (startPtr + strlen(startPtr));
+
+                    // Check if the directory has a semicolon or quotes at the end
+                    if (dirEnd > dirStart && (*(dirEnd - 1) == ';' || *(dirEnd - 1) == '"'))
+                        dirEnd--;
+
+                    // Save the current directory by creating a substring
+                    if (dirEnd > dirStart)
+                    {
+                        // Create a null-terminated copy
+                        size_t dirLen = dirEnd - dirStart;
+                        char* tempDir = (char*)malloc(dirLen + 1);
```

Comment:
Use `List<char>` or `UnownedStringSlice` instead of `malloc`.

Reply:
Okay, switched to Slang's String representation